src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.cpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace gr {
+
+KerrSchildCoords::KerrSchildCoords(const double spin_parameter) noexcept
+    : spin_parameter_(spin_parameter) {}
+
```

Comment:
Please clarify what spin parameter you use here, as this has caused a lot of confusion in the past: spin_parameter -> dimensionful_spin_parameter or spin_parameter -> dimensionless_spin_parameter

Also, is there any advantage to having this for any spin direction, instead of just spin along the z axis? Would that be hard to add to this (maybe in a separate PR)?

Reply:
Thanks for clarifying! I'm just curious: is there a reason you can't pass in a_* = \chi = a/M = S/M^2 instead, and call it dimensionless_spin? That's what's done in KerrSchild.hpp? You can always define a inside the function body if it makes the rest of the formulas easier.

Reply:
We (myself, @geoffrey4444, @markscheel, @moxcodes) would prefer this class take in two doubles: the mass and the dimensionless spin. Then check the dimensionless spin is in [-1,1]. This is more consistent with other uses of spin in spectre. This would prevent people from passing in the wrong spin by mistake (it is harder to pass in the wrong mass, because there are fewer ways of defining the mass). Mixing up different spins has caused errors in the past...

Reply:
Would you be okay moving the discussion of whether it should take `a` or `M` and `chi` to an issue? I'm not sure I agree with you and we really need to get this merged.

Reply:
Personally, I would be okay making an issue, but I'm not going to speak for others.

Reply:
I'd be fine with making it an issue as well.

Reply:
Issue is fine with me.

Reply:
That said, given the other cleanup changes being made, would this change really delay things much? 

Just pass in mass and dimensionless spin to the constructor, then compute and store as a member variable spin_a_, add an assert that abs(dimensionless_spin) < 1, then everything else stays the same.

I think it's worth getting right now if it's really as simple as I describe above, for consistency with the reset of the code (this same PR has a constructor that takes mass and dimensionless spin later on! as do other parts of spectre) and to avoid subtle bugs by passing in the wrong spin (which have happened in the past).

That said, if you ask me to approve, I'll approve, whether or not this is done. If it's not, I'll make the issue.

Reply:
It is not the magnitude of the change it is that I disagree with you, and yesterday Cris said he does too. The time it will take to agree is the issue. Thanks for understanding.

Reply:
Sounds good. I'll file an issue for us to discuss this down the road, to see if we can come to an agreement later on.